### PR TITLE
update video status

### PR DESCRIPTION
**Pull request by muhammad-ammar**
[TNL-4777](https://openedx.atlassian.net/browse/TNL-4777)

@mushtaqak This is ready for review.

### CODE REVIEW

**Comment by mushtaqak** [comment]
LGTM 👍

**Comment by muhammad-ammar** [comment]
@cahrens Do you have some time to review this PR?

**Comment by cahrens** [comment]
@muhammad-ammar I'm out sick today, and then on vacation until after the New Year (and I don't actually have any specific knowledge of video state). I'd suggest that you post in the TNL hipchat room for a reviewer.

**Comment by muhammad-ammar** [comment]
@cahrens oh..get well soon

File: edxval/api.py [hunk 1]
```diff
@@ -158,6 +158,33 @@ def update_video(video_data):
         raise ValCannotUpdateError(serializer.errors)
 
 
+def update_video_status(video_data):
```
Comment (nasthagiri) [inline edxval/api.py]: Nit: Per clean-code best practices, it's better to send only the data that this function needs as separate parameters.  So `video_id` and `status` should be separate parameters to this function.  And then, the implementation of this method need not be as defensive - that is, no longer a need to call `get` on the given `dict`.

File: edxval/models.py [hunk 2]
```diff
@@ -190,3 +194,16 @@ def content_type(self):
...
+
+@receiver(models.signals.post_save, sender=Video)
+def video_status_update_callback(sender, **kwargs):  # pylint: disable=unused-argument
+    """
+    Log video status for an existing video instance
+    """
+    video = kwargs['instance']
+
+    if kwargs['created']:
+        logger.info('Video created with id [%s] and status [%s]', video.edx_video_id, video.status)
```
Comment (nasthagiri) [inline edxval/models.py]: Nit: I find it very useful to have a common prefix for all Splunk logs within a module.  It then makes it easier to query in Splunk.  So, for example, for VAL, you may consider prefixing all log messages with "VAL: "

File: edxval/tests/test_api.py [hunk 3]
```diff
@@ -1099,3 +1100,22 @@ def test_remove_video_for_course(self):
...
+    """
+    @patch('edxval.models.logger')
+    def test_update_video_status_logging(self, mock_logger):
+        """
+        Tests video status update logging
+        """
+        video = Video.objects.create(**constants.VIDEO_DICT_FISH)
+        mock_logger.info.assert_called_with(
+            'Video created with id [%s] and status [%s]', video.edx_video_id, constants.VIDEO_DICT_FISH.get('status')
+        )
```
Comment (nasthagiri) [inline edxval/tests/test_api.py]: You may want to also have a negative test here - to make sure the "Status changed to" logging does *not* happen at the time the `Video` object is created.

Comment (muhammad-ammar) [inline edxval/tests/test_api.py]: @nasthagiri But we are already verifying that correct message is logged upon `Video` object creation. If for some reason incorrect message is logged when `Video` object is created then the above assert will surely fail. Am i missing something?

**Comment by muhammad-ammar** [comment]
@nasthagiri Addressed your feedback.

**Pull request merged**